test/lower-non-standard-types.ll

```diff
@@ -0,0 +1,49 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv -s %t.bc -o - | llvm-dis -o - | FileCheck %s --implicit-check-not="<6 x i32>"
+
+; CHECK: [[ASCastInst:%.*]] = addrspacecast <3 x i64> addrspace(1)* @Id to <3 x i64> addrspace(4)*
+; CHECK: [[LoadInst1:%.*]] = load <3 x i64>, <3 x i64> addrspace(4)* [[ASCastInst]], align 32
+; CHECK: [[ExtrElInst1:%.*]] = extractelement <3 x i64> [[LoadInst1]], i64 0
+; CHECK: [[TruncInst1:%.*]] = trunc i64 [[ExtrElInst1]] to i32
```

Comment:
I'm not sure we are doing semantically correct transformation. Originally we casted to `<6 x i32>` and took 0th and 4th elements:
```
 v       v 
|0|1|2|3|4|5|
```
With this patch we keep it `<3 x i64>` and take 0th and 2nd elements:
```
 v       v 
|0|1|2|3|4|5|
|0  |1  |2  |
```
According to the [documentation](https://llvm.org/docs/LangRef.html#trunc-to-instruction) for `trunc` instruction:
 
> The ‘trunc’ instruction truncates the **high order** bits in value and converts the remaining bits to ty2.

If I understand it correctly we get elements corresponding to 1st and 5th elements of `<6 x i32>`
```
 v       v 
|0|1|2|3|4|5|
|0  |1  |2  |
   ^       ^
```
If this is correct, in some cases we should shift the 64-bit value right for 32 bits before the truncation.